ion-c-sys/src/writer.rs

```diff
@@ -112,14 +142,27 @@ impl<'a> IonCWriterHandle<'a> {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn set_field_name(&mut self, field: &str) -> IonCResult<()>{
-        // Ion C promises that it won't do mutation for this call!
-        let mut ion_field = ION_STRING::from_str(field);
-        ionc!(ion_writer_write_field_name(self.writer, &mut ion_field))
+    pub fn start_field<'b, 'c>(&'b mut self, field: &'c str) -> IonCFieldWriterContext<'a, 'b, 'c> {
```

Comment:
Did you consider `field` or `write_field`?

The previous name made sense. The new one doesn't (to me!). Where is `end_field`, for example. What does 'start' even mean?

Reply:
I wanted to describe that the it wasn't a writing a field, but a partial application for writing a field.  I am good with `field` if that is more descriptive.

Reply:
Actually, after seeing therapon's question, I agree with the name you picked!

Reply:
Funny enough--I came around to `field` as a useful name because it technically doesn't start anything, it just partially applies the function (so to speak).  Let me know if you prefer I change it back.